src/niri.rs

```diff
@@ -906,19 +916,23 @@ impl State {
             self.niri.cursor_texture_cache.clear();
         }
 
+        let default_keyboard = config.input.default_keyboard();
+
+        let old_keyboard = old_config.input.default_keyboard();
+
         // We need &mut self to reload the xkb config, so just store it here.
-        if config.input.keyboard.xkb != old_config.input.keyboard.xkb {
-            reload_xkb = Some(config.input.keyboard.xkb.clone());
+        if default_keyboard.xkb != old_keyboard.xkb {
```

Comment:
This should check against the current keyboard I guess?

Reply:
i have no idea what i was doing here, sorting that out